### PR TITLE
Refactor substring matcher getters

### PR DESCRIPTION
**Pull request by JakeSteinebronn**
# Changes
The user cache will be used for user tag construction as well as the user search, so it's necessary to separate user cache retrieval from substring matcher retrieval. This commit does that, as well as separates the user and archived-route matchers from one another into their own hooks. 

Also, the search tab used to block view until the matchers and data were retrieved. This doesn't need to happen; instead, we can render the search bar and block *actual searching* until the data are retrieved. 
# Issue ticket number and link
Relevant to #195

### CODE REVIEW

File: screens/search/Search.tsx [hunk 1]
```diff
@@ -38,83 +39,61 @@ const Search = () => {
...
-    subMatcherData === undefined ||
-    activeRoutesData === undefined
-  ) {
-    console.error(activeRoutesError);
-    console.error(subMatcherError);
-    return null;
-  }
+  if (userMatcherQuery.isError) console.error(userMatcherQuery.error);
+  if (archivedMatcherQuery.isError) console.error(archivedMatcherQuery.error);
+  if (activeRoutesQuery.isError) console.error(activeRoutesQuery.error);
```
Comment (tylerhm) [inline screens/search/Search.tsx:96]: can we return `null` on these?

Also, why aren't we blocking when the queries are loading anymore?

Comment (JakeSteinebronn) [reply screens/search/Search.tsx:96]: Cuz it leads to a weird and unnecessary UX where you click on the search tab and it hangs for like a second or however long it takes to get the data to search on. Like I mentioned in the PR message, this doesn't need to happen, we can render the search bar and then the data fetching can block the actual searching, but the search bar should be responsive

Comment (JakeSteinebronn) [reply screens/search/Search.tsx:96]: This is also why I decided not to return `null` on these; if for whatever reason we don't get one of the matchers, well that sucks but the other parts of the search bar should remain functional

Comment (tylerhm) [reply screens/search/Search.tsx:96]: Why not render a spinner in the content below the search bar instead? We don't need to get rid of the whole spinner.

Comment (JakeSteinebronn) [reply screens/search/Search.tsx:96]: I'm confused.. what spinner did I get rid of?

Comment (tylerhm) [reply screens/search/Search.tsx:96]: Sorry, you didn't get rid of it. We never had one.

The UX just feels wrong right now, and I think the root of it is that there isn't a clear distinction between "searching" and "done searching but no results". Rendering a spinner while searching (and loading matchers), would fix this.

**Comment by tylerhm** [comment]
> Spinner now shows when results are pending (which can only happen if the queries are still fetching)

An unable to repro this behavior. Am I missing something?

https://user-images.githubusercontent.com/36250052/219411162-e395de99-a0e1-48a0-8325-48a210684b87.mp4

**Review by tylerhm — APPROVED** [review]
spoke offline and addressed UX with message.

**Pull request merged**